src/Compression/ZipFileMember.class.st

```diff
@@ -193,7 +193,7 @@ ZipFileMember >> uncompressDataTo: aStream [
 		| chunkSize |
 		chunkSize := 32768 min: readDataRemaining.
 		buffer := decoder next: chunkSize into: buffer startingAt: 1.
-		aStream next: chunkSize putAll: buffer startingAt: 1.
+		aStream next: (chunkSize min: buffer size) putAll: buffer startingAt: 1.
```

Comment:
In the case I experimented, I ended up with chunkSize of 32768 and buffer size of 20242
This lead to a SubscriptOutOfBounds error.
This is a very naive hack, but maybe someone with more knowledge may know how much sense does make that change